Hacktober Fest 2020/Problems/SPOJ/ACODE.cpp

```diff
@@ -0,0 +1,33 @@
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	char a[5010];
+    int i, j;
+    scanf("%s", a);
+    while(a[0] != '0')
+    {
+        int n = strlen(a);
+        long long int b[n];
+        for(i = 0; i < n; i++)
+            b[i] = 0;
+        b[0] = 1;
+        for(i = 1; i < n; i++)
+        {
+            j = (a[i - 1] - '0') * 10 + (a[i] - '0');
+            if(a[i] - '0')
+                b[i] = b[i - 1];
+			if(j > 9 && j < 27)
+			{
+				if(i == 1)
+					b[i] = b[i] + 1;
+				else
+					b[i] = b[i] + b[i-2];
+			}
+        }
+        printf("%lld\n", b[n - 1]);
+        scanf("%s", a);
+    }
+    if(a[0] == '0')
+	return 0;
```

Comment:
What about making the code modular? That will be easier for the learners to understand the I/O as well as the main algo.
Instead of including all the code inside the main function (which is a pretty bad standard in the industry), you may try to form some functions!

What about approaching the problem through the OOP functionalities of C++ ?